### PR TITLE
PR: Update Amazon Linux 2 SCA

### PR DESCRIPTION
**Pull request by jcruzlp**
|Related issue|
|---|
| #9541 |

## Description

This PR updates the SCA checks for Amazon Linux 2 and removes non-existing checks.

Following the tags changes described in the issue: https://github.com/wazuh/wazuh/issues/9541#issuecomment-895001414

As a future task, we should study the new SCA benchmark for CIS and add new checks to replace the outdated and removed ones.

This PR is also part of the epic #9694

### CODE REVIEW

**Review by JavierBejMen — APPROVED** [review]
LGTM

**Comment by jcruzlp** [comment]
Need to add the changes to the changelog.md file

**Review by fabamatic — DISMISSED** [review]
LGTM

**Review by JavierBejMen — APPROVED** [review]
LGTM

**Review by 72nomada — APPROVED** [review]
LGMT

File: ruleset/sca/amazon/cis_amazon_linux_2.yml [hunk 1]
```diff
@@ -928,256 +897,162 @@ checks:
       - More detailed documentation on OpenLDAP is available at https://www.openldap.org
     condition: none
     rules:
-      - 'c:systemctl is-enabled slapd ->  r:enabled'
+      - 'p:slapd'
```
Comment (chemamartinez) [inline ruleset/sca/amazon/cis_amazon_linux_2.yml:900]: Same with:
```
# rpm -q openldap-servers
```

File: ruleset/sca/amazon/cis_amazon_linux_2.yml [hunk 2]
```diff
@@ -46,46 +46,14 @@ checks:
...
-      - tsc: ["CC6.3"]
-    condition: all
-    rules:
-      - 'c:modprobe -n -v hfsplus -> r:install /bin/true|Module hfsplus not found'
-      - 'not c:lsmod -> r:hfsplus'
-
-# 1.1.1.4 squashfs: filesystem
-  - id: 20503 
+# 1.1.1.2 squashfs: filesystem
+  - id: 20503
```
Comment (chemamartinez) [inline ruleset/sca/amazon/cis_amazon_linux_2.yml:50]: We are missing IDs 20501 and 20502, if we remove checks we'd have to reassign IDs

File: ruleset/sca/amazon/cis_amazon_linux_2.yml [hunk 3]
```diff
@@ -133,10 +101,10 @@ checks:
     condition: any
     rules:
       - 'c:mount -> r:\s/tmp\s'
-      - 'c:systemctl is-enabled tmp.mount -> r:enabled'
+      - 'p:tmp.mount'
```
Comment (chemamartinez) [inline ruleset/sca/amazon/cis_amazon_linux_2.yml:104]: The CIS benchmark doesn't check for the process `tmp.mount`. Check that the title says `Ensure /tmp is configured`, so we would have to apply this check:
```
# systemctl show "tmp.mount" | grep -i unitfilestate
```

File: ruleset/sca/amazon/cis_amazon_linux_2.yml [hunk 4]
```diff
@@ -383,28 +351,28 @@ checks:
...
-      - cis: ["1.1.19"]
+      - cis: ["1.1.23"]
       - cis_csc: ["8.4","8.5"]
       - pci_dss: ["2.2.4"]
       - nist_800_53: ["CM.1"]
       - tsc: ["CC5.2"]
     condition: none
     rules:
-      - 'c:systemctl is-enabled autofs -> r:enabled'
+      - 'p:autofs'
```
Comment (chemamartinez) [inline ruleset/sca/amazon/cis_amazon_linux_2.yml:368]: Check if automounting is disabled, not running.
```
# systemctl show "autofs.service" | grep -i unitfilestate=enabled
```
See check 1.1.23 of the CIS benchmark for Amazon Linux 2 2.0.0

File: ruleset/sca/amazon/cis_amazon_linux_2.yml [hunk 5]
```diff
@@ -875,10 +844,10 @@ checks:
       - tsc: ["CC5.2"]
     condition: none
     rules:
-      - 'c:systemctl is-enabled avahi-daemon -> r:enabled'
+      - 'p: avahi-daemon'
```
Comment (chemamartinez) [inline ruleset/sca/amazon/cis_amazon_linux_2.yml:847]: We have to check if Avahi is installed, not running
```
# rpm -q avahi-autoipd avahi
```

So the rules should look like:
```yml
condition: all
rules:
  - 'c:rpm -q avahi-autoipd -> r:package avahi-autoipd is not installed'
  - 'c:rpm -q avahi -> r:package avahi is not installed'
```

File: ruleset/sca/amazon/cis_amazon_linux_2.yml [hunk 6]
```diff
@@ -891,10 +860,10 @@ checks:
       - tsc: ["CC5.2"]
     condition: none
     rules:
-      - 'c:systemctl is-enabled cups ->  r:enabled'
+      - 'p:cups'
```
Comment (chemamartinez) [inline ruleset/sca/amazon/cis_amazon_linux_2.yml:863]: Same with:
```
# rpm -q cups
```

File: ruleset/sca/amazon/cis_amazon_linux_2.yml [hunk 7]
```diff
@@ -909,11 +878,11 @@ checks:
       - More detailed documentation on DHCP is available at https://www.isc.org/software/dhcp
     condition: none
     rules:
-      - 'c:systemctl is-enabled dhcp ->  r:enabled'
+      - 'p:dhcp'
```
Comment (chemamartinez) [inline ruleset/sca/amazon/cis_amazon_linux_2.yml:881]: Same with:
```
# rpm -q dhcp
```

File: ruleset/sca/amazon/cis_amazon_linux_2.yml [hunk 8]
```diff
@@ -928,256 +897,162 @@ checks:
...
     rules:
-      - 'c:systemctl is-enabled slapd ->  r:enabled'
+      - 'p:slapd'
 
 
 
-# 2.1.7 Ensure nfs-utils is not installed or the nfs-server service is masked (Automated)
-  - id: 20551 
+# 2.1.17 Ensure nfs-utils is not installed or the nfs-server service is masked (Automated)
+  - id: 20551
```
Comment (chemamartinez) [inline ruleset/sca/amazon/cis_amazon_linux_2.yml:905]: This check doesn't match with the 2.1.17 of the benchmark (`Ensure nfs-utils is not installed or the nfs-server service is masked`). Neither do the rules.

File: ruleset/sca/amazon/cis_amazon_linux_2.yml [hunk 9]
```diff
@@ -928,256 +897,162 @@ checks:
...
-      - 'c:ystemctl is-enabled nfs ->  r:enabled'
-      - 'c:ystemctl is-enabled nfs-server ->  r:enabled'
-      - 'c:ystemctl is-enabled rpcbind ->  r:enabled'
+      - 'p:nfs'
+      - 'p:nfs-server'
+      - 'p:rpcbind'
 
-# 2.1.8 Ensure DNS Server is not enabled (Automate)
-  - id: 20552 
+# 2.1.7 Ensure DNS Server is not enabled (Automate)
```
Comment (chemamartinez) [inline ruleset/sca/amazon/cis_amazon_linux_2.yml:922]: Checks from 2.1.7 to 2.1.13 of the CIS Benchmark could be located after 20550 to follow the logical order, avoiding having the check 2.1.17 (of the benchmark) between them.

We could also add the checks 2.1.14 and 2.1.15 of the CIS benchmark to complete the family.

File: ruleset/sca/amazon/cis_amazon_linux_2.yml [hunk 10]
```diff
@@ -928,256 +897,162 @@ checks:
...
-      - cis: ["2.1.8"]
+      - cis: ["2.1.7"]
       - cis_csc: ["9.2"]
       - pci_dss: ["2.2.2"]
       - nist_800_53: ["CM.1"]
       - tsc: ["CC5.2"]
     condition: none
     rules:
-      - 'c:systemctl is-enabled named -> r:enabled'
+      - 'p:named'
```
Comment (chemamartinez) [inline ruleset/sca/amazon/cis_amazon_linux_2.yml:936]: We have to check:
```
# rpm -q bind
```

File: ruleset/sca/amazon/cis_amazon_linux_2.yml [hunk 11]
```diff
@@ -928,256 +897,162 @@ checks:
...
-      - 'c:ystemctl is-enabled rpcbind ->  r:enabled'
+      - 'p:nfs'
+      - 'p:nfs-server'
+      - 'p:rpcbind'
 
-# 2.1.8 Ensure DNS Server is not enabled (Automate)
-  - id: 20552 
+# 2.1.7 Ensure DNS Server is not enabled (Automate)
+  - id: 20552
     title: " Ensure DNS Server is not enabled "
```
Comment (chemamartinez) [inline ruleset/sca/amazon/cis_amazon_linux_2.yml:924]: Ensure DNS server is not **installed**

File: ruleset/sca/amazon/cis_amazon_linux_2.yml [hunk 12]
```diff
@@ -928,256 +897,162 @@ checks:
...
-      - cis: ["2.1.9"]
+      - cis: ["2.1.8"]
       - cis_csc: ["9.2"]
       - pci_dss: ["2.2.2"]
       - nist_800_53: ["CM.1"]
       - tsc: ["CC5.2"]
     condition: none
     rules:
-      - 'c:systemctl is-enabled vsftdp -> r:enabled'
+      - 'p:vsftdp'
```
Comment (chemamartinez) [inline ruleset/sca/amazon/cis_amazon_linux_2.yml:952]: Same here:
```
# rpm -q vsftpd
```

File: ruleset/sca/amazon/cis_amazon_linux_2.yml [hunk 13]
```diff
@@ -928,256 +897,162 @@ checks:
...
-      - cis: ["2.1.10"]
+      - cis: ["2.1.9"]
       - cis_csc: ["9.2"]
       - pci_dss: ["2.2.2"]
       - nist_800_53: ["CM.1"]
       - tsc: ["CC5.2"]
     condition: none
     rules:
-      - 'c:systemctl is-enabled httpd -> r:enabled'
+      - 'p:httpd'
```
Comment (chemamartinez) [inline ruleset/sca/amazon/cis_amazon_linux_2.yml:968]: Same here:
```
# rpm -q httpd
```

File: ruleset/sca/amazon/cis_amazon_linux_2.yml [hunk 14]
```diff
@@ -928,256 +897,162 @@ checks:
...
-      - cis: ["2.1.11"]
+      - cis: ["2.1.10"]
       - cis_csc: ["9.2"]
       - pci_dss: ["2.2.2"]
       - nist_800_53: ["CM.1"]
       - tsc: ["CC5.2"]
     condition: none
     rules:
-      - 'c:systemctl is-enabled dovecat -> r:enabled'
+      - 'p:dovecat'
```
Comment (chemamartinez) [inline ruleset/sca/amazon/cis_amazon_linux_2.yml:984]: Same here:
```
# rpm -q dovecot
```

File: ruleset/sca/amazon/cis_amazon_linux_2.yml [hunk 15]
```diff
@@ -928,256 +897,162 @@ checks:
...
-      - cis: ["2.1.12"]
+      - cis: ["2.1.11"]
       - cis_csc: ["9.2"]
       - pci_dss: ["2.2.2"]
       - nist_800_53: ["CM.1"]
       - tsc: ["CC5.2"]
     condition: none
     rules:
-      - 'c:systemctl Samba dovecat -> r:enabled'
+      - 'p:Samba'
```
Comment (chemamartinez) [inline ruleset/sca/amazon/cis_amazon_linux_2.yml:1000]: Same:
```
# rpm -q samba
```

File: ruleset/sca/amazon/cis_amazon_linux_2.yml [hunk 16]
```diff
@@ -928,256 +897,162 @@ checks:
...
-      - cis: ["2.1.13"]
+      - cis: ["2.1.12"]
       - cis_csc: ["9.2"]
       - pci_dss: ["2.2.2"]
       - nist_800_53: ["CM.1"]
       - tsc: ["CC5.2"]
     condition: none
     rules:
-      - 'c:systemctl is-enabled squid -> r:enabled'
+      - 'p:squid'
```
Comment (chemamartinez) [inline ruleset/sca/amazon/cis_amazon_linux_2.yml:1016]: Same:
```
# rpm -q squid
```

File: ruleset/sca/amazon/cis_amazon_linux_2.yml [hunk 17]
```diff
@@ -928,256 +897,162 @@ checks:
...
-      - cis: ["2.1.14"]
+      - cis: ["2.1.13"]
       - cis_csc: ["2.6",9.2"]
       - pci_dss: ["2.2.2"]
       - nist_800_53: ["CM.1"]
       - tsc: ["CC5.2"]
     condition: none
     rules:
-      - 'c:systemctl is-enabled snmpd -> r:enabled'
+      - 'p:snmpd'
```
Comment (chemamartinez) [inline ruleset/sca/amazon/cis_amazon_linux_2.yml:1032]: Same:
```
# rpm -q net-snmp
```

File: ruleset/sca/amazon/cis_amazon_linux_2.yml [hunk 18]
```diff
@@ -1601,99 +1414,53 @@ checks:
...
-    rationale: "With a default accept policy the firewall will accept any packet that is not configured to be denied. It is easier to white list acceptable usage than to black list unacceptable usage."
-    remediation: "Run the following commands to implement a default DROP policy: # iptables -P INPUT DROP   # iptables -P OUTPUT DROP      # iptables -P FORWARD DROP"
+  - id: 20593
+    title: "Ensure iptables is installed"
+    description: "iptables allows configuration of the IPv4 tables in the linux kernel and the rules stored within them. Most firewall configuration utilities operate as a front end to iptables."
+    rationale: "iptables is required for firewall management and configuration."
+    remediation: "Run the following command to install iptables: # yum install iptables"
     compliance:
-      - cis: ["3.5.1.1"]
+      - cis: ["3.5.3"]
```
Comment (chemamartinez) [inline ruleset/sca/amazon/cis_amazon_linux_2.yml:1429]: 3.5.3 is a section, the check is 3.5.3.1.1

**Pull request merged**